aws/resource_service_principal_instance_profile.go

```diff
@@ -0,0 +1,37 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/databrickslabs/terraform-provider-databricks/scim"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ResourceServicePrincipalInstanceProfile binds service principal and instance profile
+func ResourceServicePrincipalInstanceProfile() *schema.Resource {
+	r := common.NewPairID("application_id", "instance_profile_id").Schema(func(
+		m map[string]*schema.Schema) map[string]*schema.Schema {
+		m["instance_profile_id"].ValidateDiagFunc = ValidInstanceProfile
+		return m
+	}).BindResource(common.BindResource{
+		CreateContext: func(ctx context.Context, servicePrincipalID, role string, c *common.DatabricksClient) error {
+			return scim.NewServicePrincipalsAPI(ctx, c).Patch(servicePrincipalID, scim.PatchRequest("add", "roles", role))
```

Comment:
this won't work as you expect, because `servicePrincipalID` (int64) is not `application_id` (uuid). `BindResource` won't work for your use-case, unless you replace `application_id` with `service_principal_id`.

Reply:
Thank you @nfx! I changed the resource to accept `service_principal_id` instead of `application_id`, and `role` instead of `instance_profile_id`.